### DISCUSSION TITLE
Implement support for ODBC-style and other database-specific connection strings

### DISCUSSION THREAD

**Discussion by dossy** [amacneil/dbmate#436] [Feature Ideas]
It would be nice for dbmate to support specifying ODBC-style connection strings (and, database-driver specific connection string formats) rather than a fully-formed database URL, as it may be easier to hand-write the connection string rather than having to URI-encode characters in the URL by hand.

See https://github.com/amacneil/dbmate/pull/435 for some background on this.

I think adding the CLI argument `--connection-string value` and parsing the ODBC-style connection string and transforming it into a well-formed database URL, which dbmate can use, can be useful functionality for some dbmate users.

The difficulty here is that, unfortunately, PostgreSQL [connection strings](https://www.postgresql.org/docs/current/libpq-connect.html#LIBPQ-CONNSTRING) do not follow the [ODBC specification for connection strings](https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqldriverconnect-function?view=sql-server-ver16#comments).  😞 

This means having a single connection string parsing implementation won't be possible, and it'll need to be per-driver, and we'll need something like a second CLI argument `--driver value` to specify which driver the connection string is intended for.

**Comment by gregwebs** [comment]
Maybe the main problem here is just the password. One thing that worries me about all this from the code is:

```
// redactLogString attempts to redact passwords from errors
```

A separate mechanism for passing the password could be used to improve security and usability.

**Comment by gregwebs** [comment]
A validated uri requires certain fields be filled out. But It would be easier for me to use dbmate if I could just use the PG* environment variables like I do elsewhere.
